frontend/packages/component-library-styles/CONTRIBUTING.md

```diff
@@ -0,0 +1,153 @@
+- [✅ How to Get Your PR Approved Faster](#-how-to-get-your-pr-approved-faster)
+
+## 🛣️ Contribution Roadmap
+
+Here’s a quick guide to the contribution process:
+
+1. **(The most important step!) Create a thread** in the `#ask-design-system` Slack channel about your change.
+
+   - **Get approval** from @markabarrett and/or @moshebaricdo (design team).
+   - For technical details, ask @levadadenys.
+
+2. **Create a separate PR** with `@code-dot-org/component-library-styles` code changes.
+
+   - Make sure the PR follows the [Best Practices](./README.md/#best-practices).
+   - Include relevant updates to tokens, colors, typography, and styles.
+
+3. **Get it approved** by:
+
+   - ✅ @moshebaricdo and/or @markabarrett (design)
+   - ✅ @levadadenys (technical implementation)
+
+4. **Merge the PR** once approved.
+
+5. **Use it in your project → Celebrate → Collect tons of appreciation!** 🎉
+
+## 🎨 How to Add a New Style or Token
+
+Most of the styles come directly from the design system Figma file. Here's a list of what can be possibly added
+and how to do it:
+
+0. Check existing design tokens for duplicates or overlap to prevent redundancy.
+
+1. **Add the token** to the appropriate SCSS file:
+
+   - Fonts → `fonts.scss`
+   - Variables → `variables.scss`
+   - Mixins → `mixins.scss`
+
+2. Ensure the token follows existing naming conventions (e.g., `field-helper-section-white`).
+
+3. If adding a new style:
+
+   - Follow the existing structure and organization.
+
+4. **Test the token or style** across light/dark themes and RTL.
+
+5. Submit a PR following the [Contribution Roadmap](#-contribution-roadmap).
+
+## 🚦Production-ready Checklist:
+
+- Approved by the design team;
+- Defined in the appropriate SCSS file;
+- Tested in light and dark themes;
+- Verified in Storybook (if applicable).
+
+## 🎯 Coding Standards
+
+✅ Use **SCSS** for consistency.  
+✅ Follow the [Best Practices](./README.md#best-practices) section in the README.  
+✅ Use **semantic colors** from `@code-dot-org/component-library-styles/colors.scss` where possible.  
+✅ Keep token names **descriptive** and follow existing naming conventions.  
+✅ Keep file size reasonable — split large files if needed.
+
+## 🧪 How to Test Your Styles
+
+There's currently no automated testing for styles. Here's how you can test your changes manually or through other
+packages automated tests:
+
+1. **Unit Tests(Manually)** — Test RTL(right-to-left) support directly since it's part of Design System scope.
+2. **Storybook Tests** — Ensure Storybook displays the token or style correctly.
+3. **Accessibility Tests** — Run Axe tests to verify accessibility in light and dark themes.
+4. **Visual Regression Tests** — Ensure the style appears correctly across browsers and devices.
+
+➡️ See the full [Testing Guide](./README.md#testing) in the README.
+
+## 🔀 Commit Message Guidelines
+
+We follow the Conventional Commits format for all commit messages to ensure a consistent and readable git history.
```

Comment:
Would it be good to prefix all of these with some kind of design system scope?  This would help understand their context in the broader context, for example in general release PR lists like [this](https://github.com/code-dot-org/code-dot-org/releases/tag/v2025-03-19.1).

Reply:
Not sure, I think it might be redundant for packages changelogs being generated. ([example1](https://github.com/code-dot-org/code-dot-org/blob/staging/frontend/packages/component-library/CHANGELOG.md), [example2](https://github.com/code-dot-org/code-dot-org/blob/staging/frontend/packages/component-library-styles/CHANGELOG.md))

cc @stephenliang what do you think about it? Any chance we can automate it somehow that `feat(component-library/button)` will show in changelogs as `feat(button)` ?

Reply:
My point is that these are showing in the main repo changelogs and they don't make much sense there, as seen in the link.

I'm suggesting a prefix at the front of the whole message, so that instead of this:
```
fix(prettier): bump prettier to fix esm errors
```
We see something like this:
```
Design system: fix(prettier): bump prettier to fix esm errors
```
In the main repo changelog [here](https://github.com/code-dot-org/code-dot-org/releases/tag/v2025-03-19.1).

Reply:
I think it would probably make more sense to remove the component library from the main code base release entry and to have it produce its own release entry. The reason being that the component library is not "released" as part of the main repo but rather its own release cycle. When the release is scoped to the component library itself, the change log makes more sense. In fact, the release it process includes a [mechanism](https://github.com/release-it/release-it?tab=readme-ov-file#github-releases) to create a github release entry using conventional commits. Perhaps this can be something we add to the backlog?